libmamba/include/mamba/core/output.hpp

```diff
@@ -38,6 +37,11 @@ namespace mamba
             {
             }
 
+            inline FormattedString(const std::string_view& i)
```

Comment:
`inline FormattedString(std::string_view i)` string_view is a shallow view type

Reply:
sorry, could you explain why this is a problem even though we're storing a copy of the string in the member `std::string s`?

Reply:
It is not really a problem, just that `string_view` are not meant to be taken by reference.
A `string_view` is basically two pointers, a slice/view of a character array that can be on the heap/stack...
Copying it is fast as it is just a shallow copy.
Using a reference means passing a pointer to a class with these two pointers, it's unnecessary.
As of `const` `string_view` cannot modify the data they slice.

if you want to be able to pass a `std::string` wihtout copy, you can do
```cpp
inline FormattedString(std::string i) : s{std::move) {}
```
Now you have:
- `FormattedString("hello")` -> create the string;
- ` std::string str = "hello"; FormattedString(str)` -> copy `str`;
- ` std::string str = "hello"; FormattedString(std::move(str))` -> move `str`, no copy;
- ` auto func() -> std::string; FormattedString(func())` -> move create, no copy;

With any of `std::string const&`, `std::string_view`, `std::string_view const&`, you would have a copy in all these cases.

Reply:
Thanks for the explanation, changed to `FormattedString(std::string_view i)` (without the reference) in 5a80e5cd